src/main/java/org/wildfly/security/auth/server/ModifiableRealmIdentity.java

```diff
@@ -47,30 +47,13 @@
     void create() throws RealmUnavailableException;
 
     /**
-     * Replace map of credentials of this identity.  If the identity does not exist, an exception is thrown.
+     * Set the credentials of this identity.  If the identity does not exist, an exception is thrown.
+     * Any existing credential(s) are replaced/updated with the new value (in a possibly realm-specific manner).
```

Comment:
Probably still going to need some further thought e.g. I don't think we want to enforce loading all of the different password types so they can wrapped in password credentials just so we can add a public key.